src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModel.java

```diff
@@ -69,11 +70,13 @@ private void updateExercises() {
     }
     Authentication auth = authentication.get();
     if (course == null || auth == null) {
-      ExercisesTreeViewModel emptyModel =
-          new ExercisesTreeViewModel(new ArrayList<>(), new Options());
+      ExercisesTreeViewModel emptyModel = new ExercisesTreeViewModel(new ArrayList<>(),
+          new Options());
       if (course == null) {
-        emptyModel.setEmptyTextVisible(true);
+        emptyModel = new EmptyExercisesTreeViewModel();
```

Comment:
This doesn't have to be fixed now, but it's quite confusing that `new ExercisesTreeViewModel(new ArrayList<>(), new Options)` is actually different from `new EmptyExercisesTreeViewModel()`, because the first uses the empty text of the exercises tree, whereas the second one hides the tree and instead shows the separate empty text label. One option would be renaming `isEmptyTextVisible` to something else, but then `EmptyExercisesTreeViewModel` should also be renamed. Anyways, doesn't need to be fixed now